### PR TITLE
add request object to soft_assert when possible

### PR DESCRIPTION
**Pull request by dannyroberts**
This will make these emails a lot more useful by adding two features:
- Adds the request (if available) to the email; helps to know how something got triggered, where some piece of weird data comes from, etc.
- Adds ability to log the offending value (or a few values as a tuple or dict) in the email, without changing the email subject. This makes them thread correctly!

### CODE REVIEW

File: corehq/util/soft_assert/api.py [hunk 1]
```diff
@@ -9,7 +9,9 @@ def _send_message(info, backend):
         subject='Soft Assert: [{}] {}'.format(info.key[:8], info.msg),
         message=('Message: {info.msg}\n'
                  'Traceback:\n{info.traceback}\n'
-                 'Occurrences to date: {info.count}\n').format(info=info)
+                 'Request:\n{request}\n'
+                 'Occurrences to date: {info.count}\n').format(
+                info=info, request=get_request())
```
Comment (dannyroberts) [inline corehq/util/soft_assert/api.py]: (`get_request()` returns `None` if not part of a request)

**Comment by dannyroberts** [comment]
@TylerSheffels

**Comment by dannyroberts** [comment]
@benrudolph what do you think?

**Comment by TylerSheffels** [comment]
:+1: From me

**Comment by benrudolph** [comment]
awesome. i'm all for it

**Pull request merged**